test/Microsoft.Azure.WebJobs.Host.UnitTests/Indexers/FunctionIndexerTests.cs

```diff
@@ -115,6 +115,22 @@ public void IndexMethod_IfMethodReturnsTask_DoesNotThrow()
         }
 
         [Fact]
+        public void IndexMethod_IfMethodReturnsAsyncVoid_Throws()
+        {
+            // Arrange
+            IFunctionIndexCollector index = CreateDummyFunctionIndex();
```

Comment:
I added some test helpers that deal with the boiler plate code here and avoids the casts, awaits, mocks, etc. 
Check out Fails_BindingContract_Mismatch() as an example. Can you use that pattern?